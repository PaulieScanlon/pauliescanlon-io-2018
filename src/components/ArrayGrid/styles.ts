import styled from "react-emotion";

import { common } from "../../styles/";

export const ArrayGridWrapper = styled.div({
  label: "array-grid-wrapper",
  display: "flex",
  flexWrap: "wrap",
  margin: `0 -${common.padding / 2}px 0 -${common.padding / 2}px`
});
