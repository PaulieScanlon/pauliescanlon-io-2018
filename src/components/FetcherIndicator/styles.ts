import styled from "react-emotion";

export const IndicatorWrapper = styled.div({
  label: "indicator-wrapper",
  with: "100%",
  minHeight: 300
});
