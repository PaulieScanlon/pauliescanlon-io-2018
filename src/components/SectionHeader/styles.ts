import styled from "react-emotion";

export const Header = styled.div({
  label: "header",
  position: "relative",
  display: "inline-block",
  border: "none",
  overflow: "hidden"
});
