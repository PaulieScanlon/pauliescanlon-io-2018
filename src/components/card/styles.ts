import styled from 'react-emotion';

export const CardWrapper = styled.div({
  label: 'card-wrapper'
});
