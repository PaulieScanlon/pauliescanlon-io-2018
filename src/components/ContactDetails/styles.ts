import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const ContactDetailsWrapper = styled.div({
  label: "contact-details-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const ContactLink = styled.a({
  label: "contact-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
