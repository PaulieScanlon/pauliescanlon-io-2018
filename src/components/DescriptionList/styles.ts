import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";

export const DL = styled.dl({
  label: "descriptions-list"
});

export const DT = styled.dt({
  label: "descriptions-term",
  color: draculaTheme.cyan
});

export const DD = styled.dd({
  label: "descriptions-description",
  margin: "0px 0px 10px 0px",
  color: draculaTheme.comment,
  a: {
    color: `${draculaTheme.comment}!important`
  }
});
