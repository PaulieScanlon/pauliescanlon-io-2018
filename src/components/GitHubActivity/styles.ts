import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const GitHubActivityWrapper = styled.div({
  label: "git-hub-activity-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const RepoLink = styled.a({
  label: "repo-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
