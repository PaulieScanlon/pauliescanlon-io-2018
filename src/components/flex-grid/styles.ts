import styled from 'react-emotion';

// import { IPx } from '../../styles/breakpoints';
import { mq } from '../../styles/breakpoints';

interface IStyles {
  xs?: number;
  sm?: number;
}

export const FlexContainer = styled.div({
  label: 'flex-container',
  display: 'flex',
  flexWrap: 'wrap'
});

export const FlexItem = styled.div<IStyles>(
  {
    label: 'flex-item',
    position: 'relative',
    boxSizing: 'border-box'
  },
  ({ xs, sm }) => ({
    [mq[0]]: {
      border: '1px solid red',
      width: `calc(${100 / xs}% - ${0}px)`
    },
    [mq[1]]: {
      border: '1px solid blue',
      width: `calc(${100 / sm}% - ${0}px)`
    }
  })
);
