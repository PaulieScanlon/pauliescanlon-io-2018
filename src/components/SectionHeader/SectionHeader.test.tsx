import * as React from "react";
import { shallow, mount } from "enzyme";

import { SectionHeader } from "./";
import { Header } from "./styles";
import { H1 } from "../../styles/typography";

import { draculaTheme } from "../../styles/theme";

const props = {
  leading: "Lorem-ipsum",
  trailing: "()",
  display: "inline-block"
};

const shallowDefault = () => {
  return shallow(<SectionHeader {...props} />);
};

const mountDefault = () => {
  return mount(<SectionHeader {...props} />);
};

describe("<SectionHeader />", () => {
  it("renders correctly", () => {
    const wrapper = shallowDefault();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it("has both required props", () => {
    const wrapper = mountDefault();
    expect(wrapper.props()).toEqual({ ...props });
  });
  it("has a Header", () => {
    const wrapper = shallowDefault();
    const header = wrapper.find(Header);
    expect(header).toHaveLength(1);
  });
  it("has x3 H1", () => {
    const wrapper = shallowDefault();
    const h1 = wrapper.find(H1);
    expect(h1.length).toEqual(3);
  });
  it("has x3 H1 set to display inline-block", () => {
    const wrapper = shallowDefault();
    const h1s = wrapper.find(H1);
    h1s.map(h1 => {
      expect(h1.props().display).toEqual("inline-block");
    });
  });
  it("has green H1 first", () => {
    const wrapper = shallowDefault();
    const firstH1 = wrapper
      .find(H1)
      .at(0)
      .props();
    expect(firstH1.fontColour).toEqual(draculaTheme.green);
  });
  it("has pink H1 second", () => {
    const wrapper = shallowDefault();
    const secondH1 = wrapper
      .find(H1)
      .at(1)
      .props();
    expect(secondH1.fontColour).toEqual(draculaTheme.pink);
  });
  it("has cyan H1 third", () => {
    const wrapper = shallowDefault();
    const thirdH1 = wrapper
      .find(H1)
      .at(2)
      .props();
    expect(thirdH1.fontColour).toEqual(draculaTheme.cyan);
  });
});
