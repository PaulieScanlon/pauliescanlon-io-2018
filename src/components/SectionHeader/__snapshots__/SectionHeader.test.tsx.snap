// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHeader /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionHeader
    display="inline-block"
    leading="Lorem-ipsum"
    trailing="()"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <H1
          display="inline-block"
          fontColour="#50fa7b"
        >
          → 
        </H1>,
        <H1
          display="inline-block"
          fontColour="#ff79c6"
        >
          Lorem-ipsum
        </H1>,
        <H1
          display="inline-block"
          fontColour="#8be9fd"
        >
           
          ()
        </H1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "→ ",
          "display": "inline-block",
          "fontColour": "#50fa7b",
        },
        "ref": null,
        "rendered": "→ ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lorem-ipsum",
          "display": "inline-block",
          "fontColour": "#ff79c6",
        },
        "ref": null,
        "rendered": "Lorem-ipsum",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "()",
          ],
          "display": "inline-block",
          "fontColour": "#8be9fd",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "()",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <H1
            display="inline-block"
            fontColour="#50fa7b"
          >
            → 
          </H1>,
          <H1
            display="inline-block"
            fontColour="#ff79c6"
          >
            Lorem-ipsum
          </H1>,
          <H1
            display="inline-block"
            fontColour="#8be9fd"
          >
             
            ()
          </H1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "→ ",
            "display": "inline-block",
            "fontColour": "#50fa7b",
          },
          "ref": null,
          "rendered": "→ ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lorem-ipsum",
            "display": "inline-block",
            "fontColour": "#ff79c6",
          },
          "ref": null,
          "rendered": "Lorem-ipsum",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "()",
            ],
            "display": "inline-block",
            "fontColour": "#8be9fd",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "()",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
